recipes/podofo/all/patches/0001-fix-cmake-0.9.6.patch

```diff
@@ -0,0 +1,164 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -37,7 +37,7 @@ CMAKE_POLICY(SET CMP0033 NEW) # https://cmake.org/cmake/help/v3.0/policy/CMP0033
+ endif()
+ 
+ # Load modules from our source tree too
+-SET(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
++LIST(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
+ 
+ 
+ # Builds must use this CMakeLists.txt, not the one in src/ or somewhere else.
+@@ -313,73 +313,69 @@ IF(CMAKE_COMPILER_IS_GNUCXX)
+ ENDIF(CMAKE_COMPILER_IS_GNUCXX)
+ 
+ FIND_PACKAGE(ZLIB REQUIRED)
++LINK_LIBRARIES(ZLIB::ZLIB)
+ MESSAGE("Found zlib headers in ${ZLIB_INCLUDE_DIR}, library at ${ZLIB_LIBRARIES}")
+ 
+-FIND_PACKAGE(LIBCRYPTO)
+-
+-IF(LIBCRYPTO_FOUND)
++if(PODOFO_WITH_OPENSSL)
++  FIND_PACKAGE(OpenSSL REQUIRED)
+ 	SET(PODOFO_HAVE_OPENSSL TRUE)
+-	INCLUDE_DIRECTORIES(${LIBCRYPTO_INCLUDE_DIR})
+-	MESSAGE("Found OpenSSL's libCrypto headers in ${LIBCRYPTO_INCLUDE_DIR}, library at ${LIBCRYPTO_LIBRARIES}")
+-ELSE(LIBCRYPTO_FOUND)
+-	MESSAGE("OpenSSL's libCrypto not found. Encryption support will be disabled")
+-ENDIF(LIBCRYPTO_FOUND)
++	LINK_LIBRARIES(OpenSSL::OpenSSL)
++  INCLUDE(CheckCSourceCompiles)
++
++  SET(CMAKE_REQUIRED_INCLUDES ${OPENSSL_INCLUDE_DIR})
++  SET(CMAKE_REQUIRED_LIBRARIES ${OPENSSL_LIBRARIES})
++
++  CHECK_C_SOURCE_COMPILES("#include <openssl/opensslv.h>
```

Comment:
why do we need to check OpenSSL availability and/or version if we control and know it from conan?
or is it a patch you have borrowed somewhere?

Reply:
It comes from their custom Find module file. Since we now use conan generated one, we need to add those checks for podofo internal logic. We don't really control OpenSSL version, it can be overridden by donwstream.

Reply:
but I think we still can know the final graph computed (after all overrides, isn't it?)
it seems like I fail to understand the problem here, I would just pass `PODOFO_HAVE_OPENSSL_1_1` from the conanfile
okay, you can override, e.g. you build library against OpenSSL 1.1, and when you override to OpenSSL 1.0, so OpenSSL DLL changes while you consume PoDoFo DLL which was originally linked against newer DLL... what should/would happen?
@uilianries WDYT?